backend/internal/executor/identify/identifyingexecutor.go

```diff
@@ -61,18 +60,14 @@ func (i *IdentifyingExecutor) IdentifyUser(filters map[string]interface{},
 
 	userProvider := userprovider.NewUserProvider()
 	userService := userProvider.GetUserService()
-	userID, err := userService.IdentifyUser(searchableFilter)
-	if err != nil {
-		if strings.Contains(err.Error(), "user not found") {
-			logger.Debug("User not found for the provided filters")
-			execResp.Status = flowconst.ExecFailure
-			execResp.FailureReason = "User not found"
-			return nil, nil
-		}
-
-		logger.Error("Error identifying user", log.Error(err))
-		return nil, err
+	userID, svcErr := userService.IdentifyUser(searchableFilter)
+	if svcErr != nil {
+		logger.Debug("User not found for the provided filters")
```

Comment:
We need to check if the error is a user not found error before setting this. It could be any other error as well